DT_LNK requires _BSD_SOURCE or _GNU_SOURCE to be exposed
diff --git a/tools/thermal/tmon/Makefile b/tools/thermal/tmon/Makefile
index f9c52b7fa..f073f873b 100644
--- a/tools/thermal/tmon/Makefile
+++ b/tools/thermal/tmon/Makefile
@@ -8,7 +8,7 @@ BINDIR=usr/bin
 WARNFLAGS=-Wall -Wshadow -W -Wformat -Wimplicit-function-declaration -Wimplicit-int
 override CFLAGS+= $(call cc-option,-O3,-O1) ${WARNFLAGS}
 # Add "-fstack-protector" only if toolchain supports it.
-override CFLAGS+= $(call cc-option,-fstack-protector-strong)
+override CFLAGS+= $(call cc-option,-fstack-protector-strong) -D_BSD_SOURCE
 CC?= $(CROSS_COMPILE)gcc
 PKG_CONFIG?= $(CROSS_COMPILE)pkg-config
 
